build/Makefile

```diff
@@ -60,108 +50,174 @@ YQ_INSTALL_COMMAND:=curl $(CURL_INSTALL_OPTS) -o $(YQ_INSTALL_PATH) $(YQ_INSTALL
-	mkdir -p $(@D)
+PROFILE_RESOLVER_RUNNER:=oscal/src/utils/resolver-pipeline/oscal-profile-resolve.sh
+PROFILE_RESOLVER_ARGS:="uuid-method='random-xslt'"
+
+# Do not save a backup of the in-place edited file
+SED_FLAGS:=-i
+
+ifeq ($(shell uname -s),Darwin)
+	SED_FLAGS:=-i ''
+endif
+
+$(GEN_CONTENT_DIR)/%-resolved-profile_catalog.xml: $(SRC_DIR)/%_profile.xml $(GEN_XML_COPIED)
+	@mkdir -p $(@D)
 	$(PROFILE_RESOLVER_RUNNER) $(GEN_CONTENT_DIR)/$*_profile.xml $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml $(PROFILE_RESOLVER_ARGS)
-	@sed -i'' -e "s|file:$(shell realpath $(GEN_CONTENT_DIR)/$*_profile.xml)|$(shell basename $*_profile.xml)|g" $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml	
+	sed $(SED_FLAGS) -e 's|file:$(shell realpath $(GEN_CONTENT_DIR)/$*_profile.xml)|$(shell basename $*_profile.xml)|g' $(GEN_CONTENT_DIR)/$*-resolved-profile_catalog.xml	
+
+OSCAL_COMPLETE_XML_SCHEMA:=$(OSCAL_CORE_DIR)/generated/oscal_complete_schema.xsd
 
 .PHONY: validate-xml-content
-validate-xml-content: $(GEN_XML_FILES) $(OSCAL_COMPLETE_XML_SCHEMA) ## Validate XML files
-	@xmllint --schema $(OSCAL_COMPLETE_XML_SCHEMA) --noout $(GEN_XML_FILES)
+validate-xml-content: $(GEN_XML_FILES) ## Validate XML files
+	$(MAKE) -C $(OSCAL_CORE_DIR) $(subst $(OSCAL_CORE_DIR)/,,$(OSCAL_COMPLETE_XML_SCHEMA))
+	$(XMLLINT_PATH) --schema $(OSCAL_COMPLETE_XML_SCHEMA) --noout $(GEN_XML_FILES)
+
+.PHONY: clean-xml-content
+clean-xml-content: ## Clean generated XML content
+	@echo Cleaning XML content
+	rm -f $(GEN_XML_FILES)
+
+#
+# JSON Content
+#
+
+GEN_JSON_FILES:=$(subst xml,json,$(GEN_XML_FILES))
+GEN_MIN_JSON_FILES:=$(subst .json,-min.json,$(GEN_JSON_FILES))
 
 .PHONY: convert-min-json-content
 convert-min-json-content: $(GEN_MIN_JSON_FILES) ## Convert examples from OSCAL XML to JSON
 
-.SECONDEXPANSION:
-$(GEN_CONTENT_DIR)/%-min.json: $(GEN_XML_FILES) $(GEN_XML_RESOLVED_CATALOGS)
+XSLT_RUNNER:=$(OSCAL_CORE_DIR)/xslt-runner.sh
+XML_JSON_CONVERTER_XSLT:=$(OSCAL_CORE_DIR)/generated/oscal_complete_xml-to-json-converter.xsl
+
+$(GEN_CONTENT_DIR)/%-min.json: $(GEN_XML_FILES)
+	$(MAKE) -C $(OSCAL_CORE_DIR) $(subst $(OSCAL_CORE_DIR)/,,$(XML_JSON_CONVERTER_XSLT))
 	@mkdir -p $(@D)
 	$(XSLT_RUNNER) $(XML_JSON_CONVERTER_XSLT) $(GEN_CONTENT_DIR)/$(subst json,xml,$*).xml $(GEN_CONTENT_DIR)/$*-min.json
 
 .PHONY: reformat-json-content
 reformat-json-content: $(GEN_JSON_FILES) ## Format minified JSON to pretty-printed JSON
 
-$(NPM_PREFIX_DIR)/$(NPM_PKGS_DIR):
-	$(MAKE) -C oscal/build dependencies
-
-$(GEN_CONTENT_DIR)/%.json: $(GEN_CONTENT_DIR)/%-min.json
+$(GEN_CONTENT_DIR)/%.json: $(GEN_CONTENT_DIR)/%-min.json $(YQ_PATH)
 	$(JQ_PATH) . $(GEN_CONTENT_DIR)/$*-min.json > $(GEN_CONTENT_DIR)/$*.json
 
-.PHONY: checks
-checks: validate-xml-content validate-json-content validate-yaml-content ## Check all content with schema and other validation methods
+OSCAL_COMPLETE_JSON_SCHEMA:=$(OSCAL_CORE_DIR)/generated/oscal_complete_schema.json
```

Comment:
So I am trying to understand where I went wrong, is the target that will ensure `build-core-artifacts` target is triggered?

Reply:
Makefile variables do not trigger rules. The "triggering" happens on line 193 when `validate-json-content` explicitly makes the call to make the OSCAL Core's complete json schema

Reply:
Discussed in Gitter, resolving for now.